controllers/object_controls.go

```diff
@@ -2774,11 +2774,20 @@ func transformGDSContainer(obj *appsv1.DaemonSet, config *gpuv1.ClusterPolicySpe
 			if err != nil {
 				return fmt.Errorf("ERROR: failed to get destination directory for ssl key/cert config: %w", err)
 			}
-			volumeMounts, _, err := createConfigMapVolumeMounts(n, config.Driver.CertConfig.Name, destinationDir)
-			if err != nil {
-				return fmt.Errorf("ERROR: failed to create ConfigMap VolumeMounts for custom certs: %w", err)
+
+			if config.Driver.LicensingConfig.SecretName != "" {
```

Comment:
This change does not seem correct. You are adding this code in an area that does not use the NLS license. This is in the `transformGDS` method